Feat/issue397/add stratified bar chart function

kristinawlai opened the pull request:
Adding function `autoplot.summary.estimate_scr_by()` to graph stratified results of `estimate_scr_by()`

kristinawlai committed:
add new autoplot function for horizontal bar plots of estimate_scr_by results

kristinawlai committed:
Update function...currently need to use full function name

kristinawlai committed:
Update news and description

kristinawlai committed:
add ageCat to wordlist

kristinawlai committed:
Fix linting and add to _pkgdown.yaml

kristinawlai committed:
Fix linting

kristinawlai committed:
update imports

kristinawlai committed:
Fix errors

kristinawlai committed:
fix errors

kristinawlai committed:
More error fixes

kristinawlai committed:
Update imports and NAMESPACE

kristinawlai committed:
Update params

kristinawlai committed:
Update params

kristinawlai committed:
spelling correction

kristinawlai committed:
update num_cores #

kristinawlai committed:
Update params and code style

kristinawlai committed:
Begin merging plots -- add params and define "type"

kristinawlai committed:
Add different types of functions

kristinawlai committed:
Update code for adding type options and adding fill

kristinawlai committed:
document

kristinawlai committed:
Merge branch 'main' into feat/issue397/add-stratified-bar-chart-function

kristinawlai committed:
merge in Ezra scatterplot edits

kristinawlai committed:
updates

kristinawlai committed:
move barplot to own function

kristinawlai committed:
Add barplot example

kristinawlai committed:
Updates to fix xvar error (update barplot param to yvar)

kristinawlai committed:
Delete autoplot.summary.estimate_scr_by.R

kristinawlai committed:
increment version

kristinawlai committed:
update examples

kristinawlai committed:
document

kristinawlai committed:
edit example

kristinawlai committed:
Update vignette with "type" argument

kristinawlai committed:
color check

kristinawlai committed:
Remove color check... default check is informative enough

kristinawlai committed:
Add unit tests for barplots

kristinawlai committed:
Add tests

d-morrison commented on R/autoplot.summary.seroincidence.by.R:
```diff
@@ -41,22 +42,31 @@
-    ...) {
+    type,
+    ...
+) {
```
```suggestion
    ...) {
```
This change should eliminate the indentation lint (I think this is [an error in the `lintr` package](https://github.com/r-lib/lintr/issues/2830), but let's listen to it for now)

d-morrison commented on R/strat_ests_barplot.R:
```diff
@@ -0,0 +1,81 @@
+#' Barplot method for `summary.seroincidence.by` objects
+#'
+#' @param object a `summary.seroincidence.by` object (generated by applying the `summary()` method to the output of [estimate_scr_by()]).
```
```suggestion
#' @param object a `summary.seroincidence.by` object (generated by applying the 
`summary()` method to the output of [estimate_scr_by()]).
```
see `line_length_linter` annotation

d-morrison commented on R/strat_ests_barplot.R:
```diff
@@ -0,0 +1,81 @@
+    fill_lab = NULL,
+    fill_palette = NULL,
+    ...
+) {
```
```suggestion
    ...) {
```

d-morrison commented on R/strat_ests_barplot.R:
```diff
@@ -0,0 +1,81 @@
+    yvar,
+    color_var = NULL,
+    alpha = 0.7,
+    CIs = FALSE,
```
```suggestion
    CIs = FALSE, # nolint: object_name_linter
```
see https://lintr.r-lib.org/articles/lintr.html?q=nolint#excluding-only-some-linters for details; please use `# nolint` sparingly, but in this case it seems reasonable to use

d-morrison commented on inst/examples/exm-strat_ests_barplot.R:
```diff
@@ -0,0 +1,31 @@
+  sees_pop_data_pk_100
+
+curve <-
+  typhoid_curves_nostrat_100 %>%
```
```suggestion
  typhoid_curves_nostrat_100 |>
```

Please switch to the base pipe in changed files, following updated guidance from the style guide: https://style.tidyverse.org/pipes.html

d-morrison commented on tests/testthat/test-autoplot.summary.seroincidence.by.R:
```diff
@@ -51,3 +51,133 @@ test_that(
+)
+
+test_that("Invalid plot `type` specified. Please choose either 'scatter' or 'bar'.",
+  expect_snapshot(error = TRUE, {
```
```suggestion
{
```
the `expect_snapshot()` call goes inside the `{}` of `testthat(desc, {<CODE-HERE>})`; see below

d-morrison commented on tests/testthat/test-autoplot.summary.seroincidence.by.R:
```diff
@@ -51,3 +51,133 @@ test_that(
+                      type = "whisker",
+                      dodge_width = 0.1,
+                      color_var = "catchment",
+                      CI = TRUE)
```
```suggestion
 expect_snapshot(
        error = TRUE, 
        x = {
             plot1 <- autoplot(est2sum,
                      xvar = "ageCat",
                      type = "whisker",
                      dodge_width = 0.1,
                      color_var = "catchment",
                      CI = TRUE)
          }
    )
```

(indentation probably needs fixing; wrote this by hand)

d-morrison commented on tests/testthat/test-autoplot.summary.seroincidence.by.R:
```diff
@@ -51,3 +51,133 @@ test_that(
+)
+
+test_that("The variable `{yvar}` specified by argument `yvar` does not exist in `object`. Please choose a column that exists in `object`.",
+          expect_snapshot(error = TRUE, {
```
same issue as above; move `expect_snapshot()` inside the `{}`

d-morrison commented on tests/testthat/test-autoplot.summary.seroincidence.by.R:
```diff
@@ -51,3 +51,133 @@ test_that(
+                      dodge_width = 0.1,
+                      color_var = "catchment",
+                      CI = TRUE)
+  })
```
```suggestion
  }
```
the `)` here was left over from the misplaced `expect_snapshot()`

d-morrison commented on R/strat_ests_barplot.R:
```diff
@@ -0,0 +1,81 @@
+      ggplot2::geom_errorbar(ggplot2::aes(
+        xmin = .data$incidence.rate.CI.lower * 1000,
+        xmax = .data$incidence.rate.CI.upper * 1000
+      ), width = 0.2, position = ggplot2::position_dodge(width = 0.9))
```
add a `vdiffr::expect_doppelganger()` test for CIs = TRUE? In the example code, it looks like the CIs aren't actually being displayed:

https://ucd-serg.github.io/serocalculator/preview/pr409/reference/autoplot.summary.seroincidence.by.html#ref-examples

I think it might be a scope issue in the `if()` statement; you might try adding internal `()` around the logical units to check (I haven't tried this yet, so it might not be the actual issue)

d-morrison commented on R/strat_ests_barplot.R:
```diff
@@ -0,0 +1,81 @@
+
+  # Apply custom fill palette if provided
+  if (!is.null(fill_palette) && !is.null(color_var)) {
+    plot1 <- plot1 + ggplot2::scale_fill_manual(values = fill_palette)
```
test this case too?

d-morrison reviewed (changes requested):
Coming along! Please see comments in https://github.com/UCD-SERG/serocalculator/pull/409/files

kristinawlai committed:
Update R/autoplot.summary.seroincidence.by.R

Co-authored-by: Douglas Ezra Morrison <demorrison@ucdavis.edu>

kristinawlai committed:
Update R/strat_ests_barplot.R

Co-authored-by: Douglas Ezra Morrison <demorrison@ucdavis.edu>

kristinawlai committed:
Update R/strat_ests_barplot.R

Co-authored-by: Douglas Ezra Morrison <demorrison@ucdavis.edu>

kristinawlai committed:
Update R/strat_ests_barplot.R

Co-authored-by: Douglas Ezra Morrison <demorrison@ucdavis.edu>

kristinawlai committed:
Update inst/examples/exm-strat_ests_barplot.R

Co-authored-by: Douglas Ezra Morrison <demorrison@ucdavis.edu>

kristinawlai committed:
Update tests/testthat/test-autoplot.summary.seroincidence.by.R

Co-authored-by: Douglas Ezra Morrison <demorrison@ucdavis.edu>

kristinawlai committed:
Update tests/testthat/test-autoplot.summary.seroincidence.by.R

Co-authored-by: Douglas Ezra Morrison <demorrison@ucdavis.edu>

d-morrison commented on R/strat_ests_barplot.R:
```diff
@@ -0,0 +1,81 @@
+#' Barplot method for `summary.seroincidence.by` objects
+#'
+#' @param object a `summary.seroincidence.by` object (generated by applying the 
+`summary()` method to the output of [estimate_scr_by()]).
```
```suggestion
#' `summary()` method to the output of [estimate_scr_by()]).
```

kristinawlai committed:
Update test syntax and add additional tests

kristinawlai committed:
Reconcile conflicts

Merge branch 'feat/issue397/add-stratified-bar-chart-function' of https://github.com/UCD-SERG/serocalculator into feat/issue397/add-stratified-bar-chart-function

# Conflicts:
#	tests/testthat/test-autoplot.summary.seroincidence.by.R

kristinawlai committed:
Reconcile conflicts

Merge branch 'feat/issue397/add-stratified-bar-chart-function' of https://github.com/UCD-SERG/serocalculator into feat/issue397/add-stratified-bar-chart-function

# Conflicts:
#	tests/testthat/test-autoplot.summary.seroincidence.by.R

kristinawlai committed:
REsolving merge conflicts

kristinawlai committed:
update tests

kristinawlai committed:
update snapshots

kristinawlai committed:
update labels

kristinawlai committed:
update barplot

kristinawlai committed:
fix lints

kristinawlai committed:
fix lints

kristinawlai committed:
Fix lints

kristinawlai committed:
fix linting

kristinawlai committed:
Update snaps

kristinawlai committed:
fix snaps

kristinawlai committed:
update .lintr

d-morrison commented on R/strat_ests_barplot.R:
```diff
@@ -0,0 +1,86 @@
+    y = forcats::fct_rev(.data[[yvar]]),
+    x = .data$incidence.rate * 1000,
+    fill = if (!is.null(color_var)) .data[[color_var]] else NULL
+  )) +
```
When possible, please separate the `ggplot()` call and the `aes()` call, like this:

```suggestion
  plot1 <- ggplot2::ggplot(object) + 
  ggplot2::aes(
    y = forcats::fct_rev(.data[[yvar]]),
    x = .data$incidence.rate * 1000,
    fill = if (!is.null(color_var)) .data[[color_var]] else NULL
  ) +
```

kristinawlai replied:
@d-morrison When I separated `ggplot()` and `aes()` as you suggested I kept running into layers of errors from unrecognized datasets and uninherited aesthetics. I'd like to revert it to the original code, unless you have other suggestions?

d-morrison commented on R/autoplot.summary.seroincidence.by.R:
```diff
@@ -41,22 +42,30 @@
 #'     group_var = "catchment")
 #'
+#' est2sum |> autoplot(
+#'     type ="bar",
```
```suggestion
#'     type = "bar",
```

d-morrison commented on vignettes/articles/enteric_fever_example.Rmd:
```diff
@@ -270,6 +270,14 @@ est_country_agedf <- summary(est_country_age) |>
+  yvar = "ageCat",
+  color_var = "Country",
+  CIs = TRUE
+)
```
Please review https://ucd-serg.github.io/serocalculator/preview/pr409/articles/enteric_fever_example.html to make sure it looks correct after your changes.
In particular, the barplot now appears twice in a row [here](https://ucd-serg.github.io/serocalculator/preview/pr409/articles/enteric_fever_example.html#:~:text=for%20%E2%80%9Cnoise%E2%80%9D%20above.-,Finally,-%2C%20let%E2%80%99s%20visualize%20our). 
I think we can get rid of the raw ggplot code (starting on line 282).

We should probably also add `fill_palette = country_pal` to the `autoplot()` call:

```suggestion
autoplot(
  est_country_agedf,
  type = "bar",
  yvar = "ageCat",
  color_var = "Country",
  fill_palette = country_pal,
  CIs = TRUE
)
```

d-morrison reviewed (changes requested):
Congrats on getting all the checks passing! I think we're almost done, but I did catch a few more issues: https://github.com/UCD-SERG/serocalculator/pull/409/files

kristinawlai committed:
Update R/autoplot.summary.seroincidence.by.R

Co-authored-by: Douglas Ezra Morrison <demorrison@ucdavis.edu>

kristinawlai committed:
Update R/strat_ests_barplot.R

Co-authored-by: Douglas Ezra Morrison <demorrison@ucdavis.edu>

kristinawlai committed:
more fixes

kristinawlai committed:
Update fill_palette --> color_palette and ensure vignette plots follow consistent color scheme

kristinawlai committed:
Update text regarding plot method

kristinawlai committed:
Fix lints

kristinawlai committed:
Add color palette test

kristinawlai committed:
fix lints

d-morrison committed:
a little more simplification

d-morrison reviewed (approved):
Thanks, looks good to me!
